monitor/src/main/java/edu/asu/diging/monitor/web/admin/AddGroupController.java

```diff
@@ -0,0 +1,96 @@
+package edu.asu.diging.monitor.web.admin;
+
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.WebDataBinder;
+import org.springframework.web.bind.annotation.InitBinder;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import edu.asu.diging.monitor.core.exceptions.UnstorableObjectException;
+import edu.asu.diging.monitor.core.model.impl.Group;
+import edu.asu.diging.monitor.core.service.IAppManager;
+import edu.asu.diging.monitor.core.service.IGroupManager;
+import edu.asu.diging.monitor.core.service.impl.GroupValidator;
+import edu.asu.diging.monitor.web.admin.forms.AppForm;
+import edu.asu.diging.monitor.web.admin.forms.GroupForm;
+
+@Controller
+public class AddGroupController {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    @Autowired
+    private IAppManager appManager;
+
+    @Autowired
+    private IGroupManager groupManager;
+
+    @Autowired
+    private GroupValidator groupValidator;
+
+    @InitBinder
+    protected void initBinder(WebDataBinder binder) {
+        binder.addValidators(groupValidator);
+    }
+
+    @RequestMapping(value = "/admin/groups/show", method = RequestMethod.GET)
+    public String showGroups(Model model) {
+        model.addAttribute("groups", groupManager.getGroups());
+        model.addAttribute("appCount", appManager.getApps().size());
+        return "admin/groups/show";
```

Comment:
there should be a controller for each page. the `AddGroupController` should not have methods for showing all groups.